lambda/shared/src/main/scala/feral/lambda/package.scala

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2021 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feral
+
+package object lambda {
+  type Lambda[F[_], Event, Result] = (Event, Context[F]) => F[Option[Result]]
```

Comment:
Any thoughts if this should just be an alias for `Kleisli`?

Reply:
Is there any advantage to making it so? 

One very minor downside: I think making it `Kleisli` would mean, when running it, we'd generally have to wrap the event and context in two sets of parens, since it would expect a tuple instead of being a Function2.

Reply:
`Kleisli` is enriched with lots of nice methods that are also (more likely to be) stack-safe.
http://typelevel.org/cats/api/cats/data/Kleisli.html

> since it would expect a tuple instead of being a Function2
Yep, exactly my reasoning for avoiding this.

Very similar discussion in https://github.com/http4s/http4s/discussions/4846.

Reply:
Given the http4s experience and the fact that it sounds like they're going to move away from the Kleisli alias, I'm inclined to think we probably shouldn't introduce it here. But I don't have a super strong feeling either way.

Reply:
A third option is `trait Lambda[F[_], Event, Result]` that may or may not extend `Function2` 😅

Reply:
A fourth option 😆 http4s will be rewriting its middlewares to use MTL-style `Local` instead of Kleisli. Since we are taking a middleware-oriented approach here, perhaps we should do the same.
https://github.com/http4s/http4s/pull/4758

Reply:
Trying this out in https://github.com/typelevel/feral/pull/60.